WIP: Make a chrome extension for live-testing themes

DukeFerdinand opened the pull request:
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

This extension can't be shipped yet, but can be testing with a basic demo after running `build:extension` and including a local unpacked extension in the developer's browser.

**Closes:** #18 

## Summary:

This chrome extension acts as a very simple wrapper around the existing theme switcher. SUMMARY WIP

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [x] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)


## Checklist:

- [ ] My update follows the CONTRIBUTING guidelines of this project
- [ ] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Introduce a Chrome extension for live-testing themes, refactor the theme switcher component to support multiple themes, and update documentation accordingly. Add build scripts for the extension and manage external dependencies.

New Features:
- Introduce a Chrome extension for live-testing themes, allowing developers to switch themes directly in the browser.

Enhancements:
- Refactor the theme switcher component to support multiple themes and improve theme management.

Build:
- Add a build script for the Chrome extension, including external dependencies and bundling.

Documentation:
- Update documentation to reflect changes in theme properties and methods.

DukeFerdinand committed:
feat: barebones local extension development support

You can't _ship_ this extension yet, but you can test out a basic extension after running build:extension and including a local unpacked extension.

DukeFerdinand committed:
feat: explore content injection via Chrome extension manifest

DukeFerdinand committed:
feat: surface theme-selected event

There was not an event to listen for when consuming the theme switcher, so I added one!

DukeFerdinand committed:
feat: consume theme-selected event in extension

This commit adds the meat of the extension - appending `link` tags to the document head.

DukeFerdinand committed:
fix: use correct `detail` name for event config

DukeFerdinand committed:
docs: add some clarification to popup.html

Commit:
refactor: update theme application logic

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/AlaskaAirlines/auro-themeswitcher?pullRequest=19) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/AlaskaAirlines/auro-themeswitcher?pullRequest=19) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: DukeFerdinand<br/>:x: Jason Baker<br/><hr/>**Jason Baker** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/AlaskaAirlines/auro-themeswitcher?pullRequest=19) it.</sub>

DukeFerdinand committed:
fix: disable theme application in extension, never disable reset, conditionally disable apply

States weren't quite lining up when reloading checkboxes from an external state. This commit explicitly enables/disables the "apply" button at the right times, and enables a "reset" action any time.

BREAKING CHANGE: The themes selected _do not_ get applied to the Chrome extension window, as the readiness test completely destroys the view since every auro component used is themeable (results in a blank window that looks like it broke).

Commit:
docs: rebuild docs

jason-capsule42 commented:
:tada: This PR is included in version 2.0.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@aurodesignsystem/auro-themeswitcher/v/2.0.0)
- [GitHub release](https://github.com/AlaskaAirlines/auro-themeswitcher/releases/tag/v2.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
